core/src/main/java/io/questdb/std/str/FloatingDirectCharSink.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.std.str;
+
+import java.io.Closeable;
+
+import org.jetbrains.annotations.NotNull;
+
+import io.questdb.std.Chars;
+import io.questdb.std.Mutable;
+import io.questdb.std.Unsafe;
+
+public class FloatingDirectCharSink extends AbstractCharSink implements CharSequence, Closeable, Mutable {
```

Comment:
Floating in the name - is it because of var length? It seems that it doesn't

Reply:
Its floating because you maintain the memory  outside the class and the sink floats within that memory area